File: data-structures-and-algorithms-ashrafi/python/src/recursive/problem_6/problem_6.py
def multiply(a , b):
    if b == 0:
        return 0
    return a + multiply(a, b - 1)


def main():
    num_1 = 4
    num_2 = 5

    print(f"{num_1} x {num_2} = {multiply(num_1, num_2)}")


if __name__ == "__main__":
    main()

File: data-structures-and-algorithms-ashrafi/java/src/recursive/problem6/Problem6.java
public class Problem6 {
    
    public static int multiply(int a, int b){
        if (b == 0) {
            return 0;
        }
        return a + multiply(a, b - 1);
    }

    public static void main(String[] args) {
        int num1 = 5;
        int num2 = 4;

        System.out.printf("%d x %d = %d", num1, num2, multiply(num1, num2));
    }
}
